src/engine/EventManager.cxx

```diff
@@ -562,14 +556,20 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
         else if (!tileToPlace.empty() && m_placementAllowed)
         {
           // if targetObject.size > 1 it is a tile bigger than 1x1
+            // TODO: Remove the check here, there is no need for handling this differently
           if (targetObjectNodes.size() > 1 && isPointWithinMapBoundaries(targetObjectNodes))
           {
             // instead of using "nodesToPlace" which would be the origin-corner coordinate, we need to pass ALL occupied nodes for now.
-            engine.setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), tileToPlace, false);
+            //engine.setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), tileToPlace, false);
+            // TODO: If a building is selected, it still allows for line seelction.  m_nodesToPlace would contain more buildings figure out what this is.
+            //engine.map->setTileID(tileToPlace, mouseIsoCoords); // use this for now to workaround wrongly allowed line selection
+            engine.map->setTileID(tileToPlace, m_nodesToPlace);
```

Comment:
can you clarify here what is this change ? a little mixed up for me
because the previous note state """instead of using "nodesToPlace" which would be the origin-corner coordinate, we need to pass ALL occupied nodes for now"""
and here you are doing the opposite.

Reply:
this is an old note (PR is still a draft).
The new function only needs the target coordinate (origin corner) and it figures everything out in the function (gets all the nodes that will be occupied and if placement is allowed and so on. I really dislike that we had to pass all the nodes, this is logic that should happen in the function, not where it's called.
Also the fact that we had to keep "nodesToHighlight" and "nodesToPlace" the same in event manager messed up a lot and is causing the weird placement bugs we're having.
That's the first step to clean that up

Reply:
Also, I need to look at event manager further and maybe split the check logic out of setTileID so we can also call this when checking for placement, but I need to look at this more